### PR TITLE
Add yarn audit, yarn upgrade-interactive and yarn <cmd>

### PR DESCRIPTION
**Pull request by Ayc0**

### CODE REVIEW

File: yarn-completion.plugin.zsh [hunk 1]
```diff
@@ -131,8 +129,13 @@ _yc_yarn_commands_completion() {
       'team[Maintain team memberships]' \
       'unlink[Unlink a previously created symlink for a package.]' \
       'upgrade[Upgrades packages to their latest version based on the specified range.]' \
+      'upgrade --latest[Upgrades packages to their latest.]' \
```
Comment (BuonOmo) [inline yarn-completion.plugin.zsh:132]: This is only about subcommands, not their options. However, if you want to add option completion that would be a nice idea! In the meantime I don't wanna get messy with some options printed here.
```suggestion
```

File: yarn-completion.plugin.zsh [hunk 2]
```diff
@@ -131,8 +129,13 @@ _yc_yarn_commands_completion() {
...
+      'upgrade-interactive[Upgrades selected packages to their latest version based on the specified range.]' \
+      'upgrade-interactive --latest[Upgrades selected packages to their latest version.]' \
```
Comment (BuonOmo) [inline yarn-completion.plugin.zsh:134]: Same idea here...
```suggestion
```

File: yarn-completion.plugin.zsh [hunk 3]
```diff
@@ -141,9 +144,6 @@ _yc_zsh_better_yarn_completion() {
 
   # Load custom completion commands
   case "$(_yc_yarn_command)" in
-    add)
```
Comment (BuonOmo) [inline yarn-completion.plugin.zsh:144]: Why did you delete this?

**Review by BuonOmo — CHANGES_REQUESTED** [review]
Thanks for the pull request, that is always a pleasure 😄 
There are some issues for me however, if you wanna either take a look or discuss them I'll be watching!

File: yarn-completion.plugin.zsh (hunk 2 above)
Comment (Ayc0) [reply yarn-completion.plugin.zsh:134]: I know that it isn't about the available options, but this one is the option that I use the most

File: yarn-completion.plugin.zsh [hunk 4]
```diff
@@ -81,9 +81,6 @@ _yc_yarn_remove_completion() {
 }
 
 _yc_yarn_run_completion() {
-  # Only run on `yarn run ?`
-  [[ ! "$(_yc_no_of_yarn_args)" = "3" ]] && return
-
```
Comment (Ayc0) [inline yarn-completion.plugin.zsh:86]: I've removed this because with yarn, you don't have to write `yarn run <cmd>` but `yarn <cmd>`

File: yarn-completion.plugin.zsh (hunk 3 above)
Comment (Ayc0) [reply yarn-completion.plugin.zsh:144]: Because `_yc_yarn_add_completion ` looks into the `.npm` folder
which is the cache folder for npm but not necessarily for yarn and for yarn --pnp, you cannot you the cache folder anymore (as the name are like `npm-yargs-parser-10.1.0-7202265b89f7e9e9f2e5765e0fe735a905edbaa8`)

Comment (Ayc0) [reply yarn-completion.plugin.zsh:144]: I tried to fetch the npm repo instead of looking into the cache folder but I spent more than 2 hours trying to make oh-my-zsh work (I don't know how `_values` work)

(but the fetching part was working)

**Comment by Ayc0** [comment]
I'm closing this PR because it has been stale for some time now and I don't think I'll try to continue this work

**Comment by BuonOmo** [comment]
@Ayc0 got you, thanks for the try and for the clean up :)
